canonical/vanilla-framework
# Percy GHA Migration - PR Status check reflects Percy diff state

## Done

Extension of #5089 that makes the Percy PR status check a bit smarter; now the status check will actually reflect the diff state of the Percy build. If a PR has Percy diffs, it will fail, and it will pass if it has no diffs.

In the case that a PR has diffs that are then approved, you should re-run the snapshots job to update the status check to passed.

**Do not review or merge until merging of #5089**

Closes [WD-11262](https://warthogs.atlassian.net/browse/WD-11262)

## QA

- View [example PR](https://github.com/jmuzina/test-web-percy-gha-migration/pull/30) with Percy diffs 
- Verify that the [snapshots job](https://github.com/jmuzina/test-web-percy-gha-migration/actions/runs/9119796673) fails due to diffs detected
- Follow the [Percy build link](https://percy.io/bb49709b/test-vanilla-gha-migration/builds/34264788) in the snapshots job above & verify that there are unexpected diffs present.
- Verify that the [status check](https://github.com/jmuzina/test-web-percy-gha-migration/runs/25075975392) associated with the failing build reports unexpected diffs. 
- Verify that the [snapshots job](https://github.com/jmuzina/test-web-percy-gha-migration/actions/runs/9119816948) run after fixing the above diffs passes with no diffs from baseline
- Verify that the [Percy build link](https://percy.io/bb49709b/test-vanilla-gha-migration/builds/34264863) associated with the passing snapshot reports no diffs
- Verify that the [status check](https://github.com/jmuzina/test-web-percy-gha-migration/runs/25076034957) associated with the passing build reports no diffs. 

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
![image](https://github.com/canonical/vanilla-framework/assets/46915153/c0f55bd9-6f7a-40e8-bb1b-5480b37c28ea)

[WD-11262]: https://warthogs.atlassian.net/browse/WD-11262?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
